compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -7228,4 +7235,55 @@ private boolean isEndOfFieldNamesList(SyntaxKind tokenKind) {
                 return true;
         }
     }
+
+    /**
+     * Parse error type descriptor.
+     * <p>
+     * error-type-descriptor := error [error-type-param]
+     * error-type-param := < (detail-type-descriptor | inferred-type-descriptor) >
+     * detail-type-descriptor := type-descriptor
+     * inferred-type-descriptor := *
+     * </p>
+     *
+     * @return Parsed node
+     */
+    private STNode parseErrorTypeDescriptor() {
+        startContext(ParserRuleContext.ERROR_TYPE_DESCRIPTOR);
+
+        STNode errorKeywordToken = parseErrorKeyWord();
+        STNode errorTypeParamsNode, ltToken, gtToken;
+        STToken nextToken = peek();
+        STToken nextNextToken = peek(2);
+        if (nextToken.kind == SyntaxKind.LT_TOKEN || nextNextToken.kind == SyntaxKind.GT_TOKEN) {
+            ltToken = parseLTToken();
+            nextToken = peek();
+            if (nextToken.kind == SyntaxKind.ASTERISK_TOKEN) {
+                errorTypeParamsNode = consume();
+            } else {
+                errorTypeParamsNode = parseTypeDescriptor();
+            }
+            gtToken = parseGTToken();
+        } else {
+            ltToken = STNodeFactory.createEmptyNode();
+            errorTypeParamsNode = STNodeFactory.createEmptyNode();
+            gtToken = STNodeFactory.createEmptyNode();
```

Comment:
Since it is either possible to have all three tokens, or non-of the three tokens (all-or-none situation), shall we group the three tokens  `<T>` together and have it as a separate node?

Reply:
Actually my first approach is that, isn't it cost an unnecessary node?